pkg/cmd/attestation/download/download_test.go

```diff
@@ -0,0 +1,186 @@
+	}
+
+	t.Run("fetch and store attestations successfully", func(t *testing.T) {
+		err = RunDownload(&baseOpts)
+		assert.NoError(t, err)
+
+		artifact, err := artifact.NewDigestedArtifact(baseOpts.OCIClient, baseOpts.ArtifactPath, baseOpts.DigestAlgorithm)
+		require.NoError(t, err)
+
+		assert.FileExists(t, fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+
+		actualLineCount, err := countLines(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+
+		expectedLineCount := 2
+		assert.Equal(t, expectedLineCount, actualLineCount)
+
+		err = os.Remove(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+	})
+
+	t.Run("download OCI image attestations successfully", func(t *testing.T) {
+		opts := baseOpts
+		opts.ArtifactPath = "oci://ghcr.io/github/test"
+
+		err = RunDownload(&opts)
+		assert.NoError(t, err)
+
+		artifact, err := artifact.NewDigestedArtifact(opts.OCIClient, opts.ArtifactPath, opts.DigestAlgorithm)
+		require.NoError(t, err)
+
+		assert.FileExists(t, fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+
+		actualLineCount, err := countLines(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+
+		expectedLineCount := 2
+		assert.Equal(t, expectedLineCount, actualLineCount)
+
+		err = os.Remove(fmt.Sprintf("%s/%s.jsonl", tempDir, artifact.DigestWithAlg()))
+		require.NoError(t, err)
+	})
+
+	t.Run("cannot find artifact", func(t *testing.T) {
+		opts := baseOpts
+		opts.ArtifactPath = "../test/data/not-real.zip"
+
+		err := RunDownload(&opts)
+		assert.Error(t, err)
+	})
+
+	t.Run("no attestations found", func(t *testing.T) {
+		opts := baseOpts
+		opts.APIClient = api.MockClient{
+			OnGetByOwnerAndDigest: func(repo, digest string, limit int) ([]*api.Attestation, error) {
+				return nil, nil
+			},
+		}
+
+		err := RunDownload(&opts)
+		assert.NoError(t, err)
+
+		artifact, err := artifact.NewDigestedArtifact(opts.OCIClient, opts.ArtifactPath, opts.DigestAlgorithm)
+		require.NoError(t, err)
+		assert.NoFileExists(t, artifact.DigestWithAlg())
+	})
+
+	t.Run("cannot download OCI artifact", func(t *testing.T) {
+		opts := baseOpts
+		opts.ArtifactPath = "oci://ghcr.io/github/test"
+		opts.OCIClient = oci.NewReferenceFailClient()
+
+		err := RunDownload(&opts)
+		assert.Error(t, err)
+		assert.ErrorContains(t, err, "failed to digest artifact")
+	})
+
+	t.Run("with missing OCI client", func(t *testing.T) {
+		customOpts := baseOpts
+		customOpts.ArtifactPath = "oci://ghcr.io/github/test"
+		customOpts.OCIClient = nil
+		assert.Error(t, RunDownload(&customOpts))
+	})
+
+	t.Run("with missing API client", func(t *testing.T) {
+		customOpts := baseOpts
+		customOpts.APIClient = nil
+		assert.Error(t, RunDownload(&customOpts))
+	})
+}
+
+func TestCreateJSONLinesFilePath(t *testing.T) {
+	tempDir, err := os.MkdirTemp("", "gh-attestation-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
```

Comment:
Could this be simplified using the temporary directory associated with [`t *testing.T`](https://pkg.go.dev/testing#T.TempDir)?

> _TempDir returns a temporary directory for the test to use. The directory is automatically removed when the test and all its subtests complete. Each subsequent call to t.TempDir returns a unique directory; if the directory creation fails, TempDir terminates the test by calling Fatal._

Reply:
Looks like the testing TempDir function works! I've replaced the use of `os.MkdirTemp` with that.